### DISCUSSION TITLE
thunar: blacklisted paths are accessible (dbus)

### DISCUSSION THREAD

**Discussion by TestTutorials** [netblue30/firejail#4464] [Q&A]
Hi,
Propably a newb question, sorry.
So I start a firejail: `firejail --blacklist=/var`
Then in terminal type: `ls /var` and get the expected permission denied error.
So then in same terminal in sandbox I open thunar: `thunar`
Then using thunar I can navigate to the contents of /var with no issue.

Surely this should not be possible? What am I doing wrong?

**Comment by rusty-snake** [comment]
The reason is that thunar is already running. If you now start thunar in the sandbox, it will first do basic program startup like command-line parsing. Then it checks if there already a running instance and if so talk to this instance and say it to open a new window/tab at `/var`.

In order to stop this you need to add `--dbus-user=none` (or `filter`).

See also: https://github.com/netblue30/firejail/issues/3877#issuecomment-761619106.

**Reply by TestTutorials** [reply]
That works **perfectly**, thanks.
Much appreciated

**Comment by TestTutorials** [comment]
I want to make a youtube video showing how to use firejail across various popular linux distro's but I am having issues.
It works perfectly on a bare-metal install of EndeavourOS (Arch) - but not Ubuntu nor Linux Mint.

**Bug and expected behavior**
- Firefox can still talk to an already running instance of firefox and use the extensions and theme.
- Firefox should start up an run as if it is the first time it was ever run

**Reproduce**
Steps to reproduce the behavior:
1. My home directory is `dk` so: `mkdir -p /home/dk/sandbox/firefox`
2. Run in bash `firejail --private=/home/dk/sandbox/firefox --dbus-user=none firefox`.
3. Run in bash `firejail --private=/home/dk/sandbox/firefox --dbus-user=filter firefox`.

**Environment**
 - Ubuntu 20.04 LTS, Linux Mint 20.2
 - Installed 0.9.67 from github and also tried installing from sourceforge: firejail_0.9.66-apparmor_1_amd64.deb
 - The errors happened in VirtualBox as well as QEMU/KVM
 - Reset VM to a fresh Linux install of said distributions and tried the .deb package mentioned above: Output of `firejail --version` shows 0.9.66 and that DBUS is supported

**Additional context**
I also tried enabling the apparmor kernel module but made no difference. The sanbox starts up firefox and then itself shuts down.

If firefox is not already running outside a firejail sandbox then the new sandboxed firefox works as is expected inside the sandbox

**Reply by TestTutorials** [reply]
Still no response on this what I would think is an important question for many users.
